relational_operators/SelectOperator.hpp

```diff
@@ -124,7 +145,26 @@ class SelectOperator : public RelationalOperator {
         num_workorders_generated_(0),
         simple_projection_(true),
         input_relation_is_stored_(input_relation_is_stored),
-        started_(false) {}
+        placement_scheme_(input_relation.getNUMAPlacementScheme()),
+        started_(false) {
+    if (input_relation.hasPartitionScheme()) {
+      const PartitionScheme &part_scheme = input_relation.getPartitionScheme();
+      const PartitionSchemeHeader &part_scheme_header = part_scheme.getPartitionSchemeHeader();
+      int num_partitions = part_scheme_header.getNumPartitions();
```

Comment:
The type for num_partitions can be std::size_t, as it will always be non-negative.